src/WebJobs.Extensions.DurableTask/DurabilityProvider.cs

```diff
@@ -398,5 +398,15 @@ public virtual bool ValidateDelayTime(TimeSpan timespan, out string errorMessage
             errorMessage = null;
             return true;
         }
+
+        /// <summary>
+        ///  Returns true if the stored connection string, ConnectionName, matches the input DurabilityProvider ConnectionName.
+        /// </summary>
+        /// <param name="durabilityProvider">The DurabilityProvider used to check for matching connection string names.</param>
+        /// <returns>A boolean indicating whether the connection names match.</returns>
+        public virtual bool ConnectionNameMatches(DurabilityProvider durabilityProvider)
+        {
+            return this.ConnectionName.Equals(durabilityProvider.ConnectionName);
```

Comment:
This change makes the logic for comparing connection strings up to the DurabilityProvider. I believe this change should be sufficient for testing locally.